components/CompletedCheck.tsx

```diff
@@ -0,0 +1,27 @@
+import { Checkbox, CheckedState } from "tamagui";
+import { useContext } from "react";
+import { observer } from "mobx-react-lite";
+import { Check } from "@tamagui/lucide-icons";
+
+import { Todo, TodoContext } from "../stores/Todo.store";
+
+export const CompletedCheck = observer(function ({ todo }: { todo: Todo }) {
```

Comment:
See my suggestion below.
I would just pass an ID or index to the 'editTodo', and let the Store handle it.